src/editors/playgroundController.ts

```diff
@@ -17,9 +18,8 @@ const log = createLogger('editors controller');
 export default class PlaygroundController {
   _context?: vscode.ExtensionContext;
```

Comment:
Since we always assign this is the constructor I think we can lose the `?` (same with `_connectionController` and `_activeConnectionCodeLensProvider`) and that should allow us to get rid of an extra check or two.